spec/models/knight_spec.rb

```diff
@@ -0,0 +1,250 @@
+require 'rails_helper'
+
+RSpec.describe Knight, type: :model do
+  
+  before(:each) do
+    @game = FactoryGirl.build(:game)
+  end
+
+  describe '#valid_move?' do
+
+    it 'is a valid knight move' do
+      knight = Knight.new(file: 4, rank: 5)
+      valid_knight_moves = [
+        {3=>7},
+        {5=>7},
+        {2=>6},
+        {2=>4},
+        {6=>6},
+        {6=>4},
+        {3=>3},
+        {5=>3}
+      ]
+
+      valid_knight_moves.each do |position|
+        file = position.keys[0]
+        rank = position.values[0]
+        
+        result = knight.valid_move?(file, rank)
+
+        expect(result).to eq(true)
+      end
+    end
+
+    it 'is not valid when exceeding left game board edge' do
+      knight = Knight.new(file: 1, rank: 3)
+      invalid_file = 0
+      rank = 1
+
+      result = knight.valid_move?(invalid_file, rank)
+      
+      expect(result).to eq(false)
+    end
+
```

Comment:
lines 34 to 73
Since we discussed yesterday that we'll deal with the board edges test on the parent Piece model we should remove these (I had also written similar tests on the Rook_spec which I'll remove).